src/appleseed/foundation/utility/zipper.h

```diff
@@ -0,0 +1,100 @@
+
+//
+// This source file is part of appleseed.
+// Visit http://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2017 Gleb Mishchenko, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#ifndef APPLESEED_FOUNDATION_UTILITY_ZIPPER_H
+#define APPLESEED_FOUNDATION_UTILITY_ZIPPER_H
+
+// appleseed.foundation headers.
+#include "foundation/core/exceptions/exception.h"
+
+// Boost headers.
+#include "boost/filesystem/path.hpp"
+
+// Standard headers.
+#include <string>
+#include <vector>
+#include <set>
+
+namespace foundation
+{
+
+    //
```

Comment:
We always indent code inside a namespace with four spaces, *except for top-level namespaces* that encompass the whole file: indenting those would be gratuitous and would not serve any purpose.

Please remove indentation for the code inside the `namespace foundation { ... }` block (you can check how it's done in other source files).